src/script.js

```diff
@@ -22,31 +22,74 @@ export function config (filePath) {
 }
 
 export function load (runfilePath, config, logger, requirer, access) {
-  // try to load babel-register
-  try {
-    logger.log('Requiring babel-register...')
-    if (config['babel-register']) {
+  let babelEnabled = false
+  let tsEnabled = false
+  let hasJsRunfile = false
+  let hasTsRunfile = false
+  let importError = null
+
+  if (config['babel-register']) {
+    try {
       requirer(config['babel-register'])
-    } else {
+      babelEnabled = true
+    } catch (error) {
+      importError = error
+    }
+  } else if (config['ts-register']) {
+    try {
+      requirer(config['ts-register'])
+      tsEnabled = true
+    } catch (error) {
+      importError = error
+    }
+  } else {
+    try {
       requirer('./node_modules/babel-register')
+      babelEnabled = true
+    } catch (error) {
+      importError = error
     }
-  } catch (error) {
-    logger.log('Requiring failed. Fallback to pure node.')
-    if (config['babel-register']) {
-      throw error
+
+    if (!babelEnabled) {
```

Comment:
I do think that depending on try-catch blocks are a bit hacky. But I also miss the 'automagical' transpiler detection.

The way you describe is really the way to go. Little configuration with some defaults. In the particular case of ts-node/register we can check the extension of the runfile, if it turns out to be .ts then we can try to load the default `ts-node/register`.

So first of all it would look for the runfile, fail if it is not detected. If it is a ts file it would try to load `ts-node/register` or the given transpiler.

Maybe I'm overthinking and the simple config section is the way to go. What do you think?